include/openPMD/IO/ADIOS/ADIOS1Auxiliary.hpp

```diff
@@ -87,24 +90,86 @@ getBP1DataType(Datatype dtype)
         case DT::VEC_UCHAR:
         case DT::BOOL:
             return adios_unsigned_byte;
-        case DT::INT16:
-        case DT::VEC_INT16:
-            return adios_short;
-        case DT::INT32:
-        case DT::VEC_INT32:
-            return adios_integer;
-        case DT::INT64:
-        case DT::VEC_INT64:
-            return adios_long;
-        case DT::UINT16:
-        case DT::VEC_UINT16:
-            return adios_unsigned_short;
-        case DT::UINT32:
-        case DT::VEC_UINT32:
-            return adios_unsigned_integer;
-        case DT::UINT64:
-        case DT::VEC_UINT64:
-            return adios_unsigned_long;
+        case DT::SHORT:
+        case DT::VEC_SHORT:
+            if( sizeof(short) == 2u )
+                return adios_short;
+            else if( sizeof(short) == 4u )
+                return adios_integer;
+            else if( sizeof(long) == 8u )
+                return adios_long;
+            else
+                throw unsupported_data_error("No native equivalent for Datatype::SHORT found.");
+        case DT::INT:
+        case DT::VEC_INT:
+            if( sizeof(int) == 2u )
+                return adios_short;
+            else if( sizeof(int) == 4u )
+                return adios_integer;
+            else if( sizeof(int) == 8u )
+                return adios_long;
+            else
+                throw unsupported_data_error("No native equivalent for Datatype::INT found.");
+        case DT::LONG:
+        case DT::VEC_LONG:
+            if( sizeof(long) == 2u )
+                return adios_short;
+            else if( sizeof(long) == 4u )
+                return adios_integer;
+            else if( sizeof(long) == 8u )
+                return adios_long;
+            else
+                throw unsupported_data_error("No native equivalent for Datatype::LONG found.");
+        case DT::LONGLONG:
+        case DT::VEC_LONGLONG:
+            if( sizeof(long long) == 2u )
+                return adios_short;
+            else if( sizeof(long long) == 4u )
+                return adios_integer;
+            else if( sizeof(long long) == 8u )
+                return adios_long;
+            else
+                throw unsupported_data_error("No native equivalent for Datatype::LONGLONG found.");
```

Comment:
`long long` is at least  Bit, making the
```cpp
if( sizeof(long long) == 2u )

if( sizeof(long long) == 4u )
```
branches dead code.
[C++ fundamental types](https://en.cppreference.com/w/cpp/language/types)